### PR TITLE
Changes for Open Source PDT review.

### PR DESCRIPTION
**Pull request by ichbinblau**
1. Upgrade to Ubuntu version 16.04; 
2. Disable updating sources.list

You may want to verify it by the commands below:
$ sudo docker pull xshan1/smarthome-demo:sourceslist
$ sudo docker run -d -e proxy=<ip of the proxy> --name portal -p 3030:3000 -p 3031:4000 xshan1/smarthome-demo:sourceslist

Signed-off-by: Theresa <theresa.shan@intel.com>

### CODE REVIEW

File: smarthome-web-portal/Dockerfile [hunk 1]
```diff
@@ -12,43 +12,30 @@ RUN echo $https_proxy
...
 RUN echo $http_proxy
 
-# update apt source.list
-COPY tools/docker/sources.list /etc/apt
+# Remove the comments below if you want to use a faster repo.
+# update apt source.list if it exists as tools/docker/sources.list
+#COPY tools/docker/sources.list /etc/apt
 
-RUN rm /var/lib/apt/lists/* -vf
+#RUN rm /var/lib/apt/lists/* -vf
```
Comment (gvancuts) [inline smarthome-web-portal/Dockerfile]: Can you group that line with the one above?

File: smarthome-web-portal/Dockerfile [hunk 2]
```diff
@@ -12,43 +12,30 @@ RUN echo $https_proxy
 
 RUN echo $http_proxy
 
-# update apt source.list
-COPY tools/docker/sources.list /etc/apt
+# Remove the comments below if you want to use a faster repo.
```
Comment (gvancuts) [inline smarthome-web-portal/Dockerfile]: Can you rephrase this a bit to mention it allows to use a mirror repo that's closer to them? My suggestion would be:
```
# Uncomment the two lines below if you wish to use an Ubuntu mirror repository
# that is closer to you (and hence faster). The 'sources.list' file in the tree points
# at one of the official Ubuntu mirror in Taiwan. You should update this file based
# your own location. For a list of official Ubuntu mirror repositories, check out:
# https://launchpad.net/ubuntu/+archivemirrors

File: smarthome-web-portal/initial_db.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ def hash_str(raw_str):
 DEFAULT_GATEWAYS = [
     {
         'name': 'lab',
-        'url': 'https://10.239.76.66:8000/',
+        'url': 'https://10.239.76.41:8000/',
```
Comment (gvancuts) [inline smarthome-web-portal/initial_db.py]: That IP address is not accessible publicly, how about using a more generic one (such as 172.17.0.2 which would be the one assigned to the first Docker started on a local machine?

File: smarthome-web-portal/tools/docker/README.md [hunk 4]
```diff
@@ -12,8 +12,17 @@ Here are the steps to run the Smart Home Web Portal in a Docker container.
     * Get in the `smarthome-web-portal` portal: `$ cd SmartHome-Demo/smarthome-web-portal`
 
 2. Configurations
-    * Update the preferred source in source.list
-      eg. change `archive.ubuntu.com` to `hk.archive.ubuntu.com`
+    * Choose a faster mirror by updating `sources.list` (optional)
```
Comment (gvancuts) [inline smarthome-web-portal/tools/docker/README.md]: Similar comment than above, can you rephrase that to indicate a use could choose a mirror that's closer to him?

File: smarthome-web-portal/tools/docker/README.md [hunk 5]
```diff
@@ -12,8 +12,17 @@ Here are the steps to run the Smart Home Web Portal in a Docker container.
...
+        - Remove the commented section below in the Dockerfile:
+        ```
```
Comment (gvancuts) [inline smarthome-web-portal/tools/docker/README.md]: Replace that block with whatever update you make in the Dockerfile :-)

File: smarthome-web-portal/tools/docker/README.md [hunk 6]
```diff
@@ -12,8 +12,17 @@ Here are the steps to run the Smart Home Web Portal in a Docker container.
...
+        # Remove the comments below if you want to use a faster repo.
+        # update apt source.list if it exists as tools/docker/sources.list
+        #COPY tools/docker/sources.list /etc/apt
+
+        #RUN rm /var/lib/apt/lists/* -vf
+        ```
+        - Update `tools/docker/sources.list` file
+        eg. change `free.nchc.org.tw` to the closest [mirror site](https://launchpad.net/ubuntu/+mirror/free.nchc.org.tw-archive)
```
Comment (gvancuts) [inline smarthome-web-portal/tools/docker/README.md]: I think it would be better to just give the official list of mirrors (not a specific one)

File: smarthome-web-portal/Dockerfile (hunk 2 above)
Comment (ichbinblau) [inline smarthome-web-portal/Dockerfile]: It looks much better. 👍 May I know what the tree point is?

File: smarthome-web-portal/Dockerfile (hunk 1 above)
Comment (ichbinblau) [inline smarthome-web-portal/Dockerfile]: Will do

File: smarthome-web-portal/initial_db.py (hunk 3 above)
Comment (ichbinblau) [inline smarthome-web-portal/initial_db.py]: Will do

File: smarthome-web-portal/tools/docker/README.md (hunk 4 above)
Comment (ichbinblau) [inline smarthome-web-portal/tools/docker/README.md]: Okay

File: smarthome-web-portal/tools/docker/README.md (hunk 5 above)
Comment (ichbinblau) [inline smarthome-web-portal/tools/docker/README.md]: Will do

File: smarthome-web-portal/tools/docker/README.md (hunk 6 above)
Comment (ichbinblau) [inline smarthome-web-portal/tools/docker/README.md]: Good catch. It is a copy-and-paste mistake.

File: smarthome-web-portal/Dockerfile (hunk 2 above)
Comment (gvancuts) [inline smarthome-web-portal/Dockerfile]: It's a very poor phrasing on my side :-)

How about:
```
The 'sources.list' file inside the 'tools/docker/' folder is set to use one of Ubuntu's official mirror in Taiwan.
```

Comment (ichbinblau) [inline smarthome-web-portal/Dockerfile]: Yes. This version is more easily understood for non-native speakers.

**Comment by ichbinblau** [comment]
Hi, @gvancuts. All the updates have been made. PTAL.

File: smarthome-web-portal/Dockerfile (hunk 2 above)
Comment (gvancuts) [inline smarthome-web-portal/Dockerfile]: ... including me (I had to re-read my own sentence a couple of times to figure out what I had meant) :-))

File: smarthome-web-portal/Dockerfile [hunk 7]
```diff
@@ -12,43 +12,33 @@ RUN echo $https_proxy
...
-# update apt source.list
-COPY tools/docker/sources.list /etc/apt
-
-RUN rm /var/lib/apt/lists/* -vf
+# Uncomment the two lines below if you wish to use an Ubuntu mirror repository
+# that is closer to you (and hence faster). The 'sources.list' file inside the
+# 'tools/docker/' folder is set to use one of Ubuntu's official mirror in Taiwan.
+# You should update this file based on your own location. For a list of official
+# Ubuntu mirror repositories, check out: https://launchpad.net/ubuntu/+archivemirrors
+# update apt source.list if it exists as tools/docker/sources.list
```
Comment (gvancuts) [inline smarthome-web-portal/Dockerfile]: This line is now merely repeating things said above. Can you remove it?

File: smarthome-web-portal/initial_db.py [hunk 8]
```diff
@@ -23,7 +23,7 @@ def hash_str(raw_str):
 DEFAULT_GATEWAYS = [
     {
         'name': 'lab',
-        'url': 'https://10.239.76.66:8000/',
+        'url': 'https://172.17.0.2:8000/',
```
Comment (gvancuts) [inline smarthome-web-portal/initial_db.py]: Can you make this http (instead of https). The container does not use https by default (it can be told to but I prefer we stick with the default options in our example).

File: smarthome-web-portal/tools/docker/README.md [hunk 9]
```diff
@@ -12,8 +12,20 @@ Here are the steps to run the Smart Home Web Portal in a Docker container.
...
-      eg. change `archive.ubuntu.com` to `hk.archive.ubuntu.com`
+    * Choose an Ubuntu mirror that is close to you (optional)
+        - Uncomment the last two lines below in the Dockerfile:
+        ```
+        # Uncomment the two lines below if you wish to use an Ubuntu mirror repository
+        # that is closer to you (and hence faster). The 'sources.list' file inside the
+        # 'tools/docker/' folder is set to use one of Ubuntu's official mirror in Taiwan.
+        # You should update this file based on your own location. For a list of official
+        # Ubuntu mirror repositories, check out: https://launchpad.net/ubuntu/+archivemirrors
+        # update apt source.list if it exists as tools/docker/sources.list
```
Comment (gvancuts) [inline smarthome-web-portal/tools/docker/README.md]: Delete this line (as above)

**Comment by ichbinblau** [comment]
@gvancuts . Done!

**Comment by gvancuts** [comment]
I have just done a quick sanity test over here... :+1:
